dist/svg.js

```diff
@@ -553,16 +553,28 @@ SVG.extend(SVG.PointArray, {
     // if already is an array, no need to parse it
     if (Array.isArray(array)) return array
 
-    // split points
-    array = this.split(array)
 
     // parse points
-    for (var i = 0, il = array.length, p, points = []; i < il; i++) {
-      p = array[i].split(',')
-      points.push([parseFloat(p[0]), parseFloat(p[1])])
-    }
+    var points = array.split(/\s|,/)
+
+    // validate points - https://svgwg.org/svg2-draft/shapes.html#DataTypePoints
+    // Odd number of coordinates is an error. In such cases, drop the last odd coordinate.
+    if (points.length % 2 !== 0) points.pop()
+
+    // parse points as floats
+    points = points.map(function(x) { return parseFloat(x) })
+
+    // wrap points in two-tuples
+    points = points.map(wrapPoints()).filter(function(x) { return x })
```

Comment:
Albeit it looks clean, this code is not very understandable. Its a bit magic going on here.
My idea would be `reduce` or a simple `for loop`. We lately had trouble with performance in svg.js where filter was heavily used.
A loop is still faster than a function call.